### DISCUSSION TITLE
Nest.js Microservice with Encore

### DISCUSSION THREAD

**Discussion by Lenghak** [encoredev/encore#1630] [General]
Hi! I was wondering how can we leverage the features of Nest.js to effectively build and manage microservices within the Encore framework? What are some best practices for ensuring scalability and maintainability in such an architecture, and how do Nest.js's built-in features, like dependency injection and modularity, enhance this process?

**Comment by marcuskohlberg** [comment]
You can if you want to: https://encore.dev/docs/ts/how-to/nestjs

**Comment by vincentsong** [comment]
I really love the awesome http and database performance which encore.ts provides, meanwhile I also love the nestjs's brilliant decorators, so is it possible to keep the controllers of nestjs and mix with your query validators, but only replace the express? And also expect to leverage the GraphQL Module in nestjs app, so we don't need to build a raw API endpoint.

**Reply by marcuskohlberg** [reply]
Try it and ask for help on Discord (https://encore.dev/discord) if you run into issues!
